zerver/lib/outgoing_webhook.py

```diff
@@ -38,28 +39,30 @@ def __init__(self, base_url, token, user_profile, service_name):
     # request_kwargs. It provides rest operation related info.
     # request_data is a dictionary whose format can vary depending on the type of webhook service.
     def process_event(self, event):
-        # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Dict[str, Any]]
+        # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Any]
         raise NotImplementedError()
 
-    # Given a successful response to the outgoing webhook REST operation, returns the message
-    # that should be sent back to the user.
+    # Given a successful response to the outgoing webhook REST operation, determines if response is required
+    # to be sent to user and in such case, returns the message that should be sent back.
     #
     # The response will be the response object obtained from REST operation.
     # The event will be the same as the input to process_command.
-    # The returned message will be a dictionary which should have "response_message" as key and response message to
-    # be sent to user as value.
+    # The tuple (response_required, response_message) will be returned, where
+    # 1. response_required would be boolean True or False.
+    # 2. response_message would be the response message to be sent to user. It can also be None.
```

Comment:
First of all, sorry to go back and forth on this.  I'm used to the golang convention where functions often return a tuple like `(string, bool)` or `(string, error)`, but this is because golang doesn't have `None` or exceptions, and @timabbott 's suggestion makes more sense here.

Anyway, if `response_message` can be `None`, and you can use whether or not the message is `None` to determine whether or not a response should be sent, then is it still necessary to include `response_required` in the return value?